src/components/ToolMenu/LayerTree/index.tsx

```diff
@@ -51,13 +52,11 @@ export const LayerTree: React.FC<LayerTreeProps> = ({
 
   const treeFilterFunction = (layer: OlLayer<OlSource> | OlLayerGroup) => {
 
-    // @ts-ignore
-    if (layer.getLayers) {
+    if ((layer as OlLayerGroup).getLayers) {
       return !layer.get('hideInLayerTree');
     }
 
-    // @ts-ignore
-    if (layer.getSource && layer.getSource().forEachFeature) {
+    if ((layer as OlLayer).getSource && ((layer as OlLayer).getSource() as OlSourceVector)?.forEachFeature) {
```

Comment:
Mabye you can make use of the `ol-util/LayerUtil` and one of its TypeGuards... and add a TypeGuard fot the LayerGroup ,too. :)

Reply:
Good idea, will handle it in a follow up, see #575.